src/runtime/internal/Component.ts

```diff
@@ -96,13 +97,23 @@ function make_dirty(component, i) {
 	component.$$.dirty[(i / 31) | 0] |= (1 << (i % 31));
 }
 
+
+export function add_css_to_component(component, add_css, options) {
+	component.$$ = {
+		customStyleTag: options.customStyleTag || current_component && current_component.$$.customStyleTag
+	};
+	add_css(component.$$.customStyleTag || document.head);
+}
+
 export function init(component, options, instance, create_fragment, not_equal, props, dirty = [-1]) {
 	const parent_component = current_component;
 	set_current_component(component);
 
 	const prop_values = options.props || {};
 
 	const $$: T$$ = component.$$ = {
+		...component.$$,
```

Comment:
i understand that doing this was to make the code size for the base case unchanged.
but i'm ambivalent of doing this.

what do you think @Conduitry

Reply:
I think the only option without additional code for the base output is to rewrite the function `add_css_to_component` to this:
```
export function add_css_to_component(component: any, add_css, options) {
	component.customStyleTag = options.customStyleTag || current_component && current_component.customStyleTag;
	add_css(component.customStyleTag || document.head);
}
```

But then it is a "hidden" property on the class itself.
I chose the option to include it in the $$ context object because that's why the context object exists.

Reply:
another solition could be to save the information in a variable like the current_component:

```
let appendStylesTo = document.head

export function add_css_to_component(component: any, add_css, options) {
   if (options.customStyleTag) appendStylesTo = options.customStyleTag
        add_css(appendStylesTo);
}
```